snuba/manual_jobs/__init__.py

```diff
@@ -65,6 +61,10 @@ def config_key(cls) -> str:
     def get_from_name(cls, name: str) -> "Job":
         return cast("Job", cls.class_from_name(name))
 
+    @classmethod
+    def async_finished_check(cls, job_id: str) -> bool:
+        return True
```

Comment:
I think this should raise an exception (`NotImplemented` or something) so that we don't accidentally have jobs transitioning from ASYNC_RUNNING_BACKGROUND to FINISHED because somebody forgets to override